### ISSUE TITLE
Implicit conversion loses integer precision: 'NSUInteger' (aka 'unsigned long') to 'unsigned int'

### ISSUE THREAD

**Issue by avenantsoftware** [iospirit/HIDRemote#2]
In HIDRemote.m

this part:

```
 - (unsigned)activeRemoteControlCount
{
    return ([_serviceAttribMap count]);
}
```

**Comment by felix-schwarz** [comment]
This only affects 64-Bit builds and is caused by Apple switching -[NSArray count] from

– (unsigned)count;

to

@property(readonly) NSUInteger count;

NSUInteger is identical to unsigned on 32 Bit systems, but different on 64 Bit systems. A simple cast to (unsigned) in front of [_serviceAttribMap count] should silence this warning and has no practical side effects, since the Apple IR Receiver can only distinguish 255 different remotes. I'll in fact adopt this in the next HIDRemote release.

I'd love to simply switch over to NSUInteger, but that type is only available since OS X 10.5, whereas HIDRemote also needs to support OS X 10.4 for historic reasons.

**Comment by avenantsoftware** [comment]
thanx that fixed the warning

**Comment by felix-schwarz** [comment]
This issue is fixed in 33a32ab6136feb8b8220e99ec52c7504c3a82242
